pennylane/drawer/tape_text.py

```diff
@@ -88,21 +88,25 @@ def _add_cond_grouping_symbols(op, layer_str, config):
     mapped_wires = [config.wire_map[w] for w in op.wires]
     mapped_bits = [config.bit_map[m] for m in op.meas_val.measurements]
     max_w = max(mapped_wires)
-    max_b = max(mapped_bits) + n_wires
+    max_b = max(mapped_bits)
 
-    ctrl_symbol = "╩" if config.cur_layer != config.cwire_layers[max(mapped_bits)][-1] else "╝"
-    layer_str[max_b] = f"═{ctrl_symbol}"
+    ctrl_symbol = "╩"
+    if any(config.cur_layer == stretch[-1] for stretch in config.cwire_layers[max_b]):
+        ctrl_symbol = "╝"
+    layer_str[max_b + n_wires] = f"═{ctrl_symbol}"
```

Comment:
Instead of checking whether the current layer (`config.cur_layer`) is (un)equal to the last entry of a single list of layer indices, we need to check whether it is (un)equal to any (all) of the last entries of multiple lists of layer indices (the different stretches over which the same classical wire is being used)